libs/langchain-core/src/_standard/message.ts

```diff
@@ -0,0 +1,1119 @@
+ *   id: "msg-123",
+ *   name: "user",
+ *   type: "ai",
+ *   content: [
+ *     { type: "text", text: "Hello!" },
+ *     {
+ *       type: "tool_call",
+ *       name: "search",
+ *       args: { query: "What is the capital of France?" }
+ *     }
+ *   ]
+ * };
+ * ```
+ */
+export interface Message<
+  TStructure extends $MessageStructure = $StandardMessageStructure,
+  TRole extends $MessageType = $MessageType
+> {
+  /** The message type/role, determines the content structure and available properties */
+  readonly type: TRole;
+  /** Unique identifier for this message */
+  id: string;
+  /** Optional name/identifier for the entity that created this message */
+  name?: string;
+  /** Array of content blocks that make up the message content, typed based on the structure and role */
+  content: Array<$InferMessageContent<TStructure, TRole>>;
+}
+
+/** Parameters for creating an AIMessage */
+type AIMessageParams<
+  TStructure extends $MessageStructure = $StandardMessageStructure
+> = {
+  /** Optional unique identifier for the message */
+  id?: string;
+  /** Optional name/identifier for the AI that generated this message */
+  name?: string;
+  /** Optional metadata about the AI model response (model provider, model name, etc.) */
+  responseMetadata?: $InferMessageProperty<
+    TStructure,
+    "ai",
+    "responseMetadata"
+  >;
+  /** Optional usage statistics for the AI response (token counts, etc.) */
+  usageMetadata?: $InferMessageProperty<TStructure, "ai", "usageMetadata">;
+} & (
+  | {
+      /** Array of content blocks that make up the message */
+      content: Array<$InferMessageContent<TStructure, "ai">>;
+    }
+  | {
+      /** Text content for the message */
+      content: string;
+    }
+);
+
+/**
+ * Represents a message from an AI assistant or model.
+ *
+ * This class implements the Message interface specifically for AI-generated content,
+ * providing type-safe access to AI-specific properties like response metadata and usage statistics.
+ *
+ * @template TStructure - The message structure type that defines the content and property types.
+ *                        Defaults to $StandardMessageStructure.
+ *
+ * @example
+ * ```ts
+ * // Create an AI message with simple text
+ * const aiMessage = new AIMessage("Hello, how can I help you?");
+ *
+ * // Access the combined text content
+ * console.log(aiMessage.text); // "Hello, how can I help you?"
+ *
+ * // Create an AI message with structured content
+ * const aiMessageWithContent = new AIMessage([
+ *   { type: "text", text: "Here's the answer: " },
+ *   { type: "text", text: "42" }
+ * ]);
+ *
+ * // Access the combined text content
+ * console.log(aiMessageWithContent.text); // "Here's the answer: 42"
+ * ```
+ */
+export class AIMessage<
+  TStructure extends $MessageStructure = $StandardMessageStructure
+> implements Message<TStructure, "ai">, $BrandedMessage
+{
+  /** @internal */
+  [__BRANDED_MESSAGE]: never;
```

Comment:
_nit_: I think we should try naming this internal property differently so that it's either searchable or explainable at first glance, say if I do this:

```typescript
const system: SystemMessage = {
  type: "system",
  id: "123",
  content: [{ type: "text", text: "hello" }],
  text: "hello",
};
```

I'm getting the following error, which is a bit confusing:

```
Property '[__BRANDED_MESSAGE]' is missing in type '{ type: "system"; id: string; content: { type: "text"; text: string; }[]; text: string; }' but required in type 'SystemMessage<$StandardMessageStructure>'.
```